src/audio_worklet_node.rs

```diff
@@ -0,0 +1,379 @@
+use crate::utils::float_buffer_to_js;
+use crate::*;
+
+use crossbeam_channel::{self, Receiver, Sender};
+use napi::*;
+use napi_derive::js_function;
+use web_audio_api::node::*;
+use web_audio_api::worklet::*;
+use web_audio_api::AudioParamDescriptor;
+
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex, OnceLock};
+
+/// Rust to JS processor inputs
+pub(crate) struct ProcessorArguments {
+    // processor inputs (unsafely cast to static)
+    inputs: &'static [&'static [&'static [f32]]],
+    // processor ouputs (unsafely cast to static)
+    outputs: &'static [&'static [&'static [f32]]],
+    // processor audio params (unsafely cast to static)
+    params: Vec<(&'static str, &'static [f32])>,
+    // AudioWorkletGlobalScope currentTime
+    current_time: f64,
+    // AudioWorkletGlobalScope currentFrame
+    current_frame: u64,
+    // AudioWorkletGlobalScope sampleRate
+    sample_rate: f32,
+    // tail_time return value
+    tail_time: Arc<AtomicBool>,
+}
+
+unsafe impl Send for ProcessorArguments {}
+
+// channel from main to worker
+#[allow(clippy::type_complexity)] // will refactor later
+pub(crate) fn send_recv_pair() -> &'static Mutex<(
+    Option<Sender<ProcessorArguments>>,
+    Option<Receiver<ProcessorArguments>>,
+)> {
+    static PAIR: OnceLock<
+        Mutex<(
+            Option<Sender<ProcessorArguments>>,
+            Option<Receiver<ProcessorArguments>>,
+        )>,
+    > = OnceLock::new();
+    PAIR.get_or_init(|| {
+        let (send, recv) = crossbeam_channel::unbounded();
+        Mutex::new((Some(send), Some(recv)))
+    })
+}
+
+pub(crate) struct SendItem2(Vec<AudioParamDescriptor>);
+
+// channel from worker to main
+#[allow(clippy::type_complexity)] // will refactor later
+pub(crate) fn send_recv_pair2() -> &'static (Sender<SendItem2>, Receiver<SendItem2>) {
```

Comment:
The Sender must be wrapped in a Mutex to avoid mix-ups in concurrent access (not entirely sure if that is possible from nodejs, but let's keep it safe)
The MutexGuard must be held from the `send` call to after the  `AudioWorkletNode::new` call